packages/core/src/report-components/table-data.ts

```diff
@@ -0,0 +1,15 @@
+import { TaskMetaData, DependencyResult, ReportComponentType } from '../types';
+import { ReportComponentData } from './report-component-data';
+
+export default class TableData extends ReportComponentData {
+  taskResult: DependencyResult[];
+
+  constructor(
+    meta: TaskMetaData,
+    taskResult: DependencyResult[],
+    reportComponentType: ReportComponentType = ReportComponentType.Table
+  ) {
+    super(meta, reportComponentType);
+    this.taskResult = taskResult;
+  }
```

Comment:
If you slightly reorder the params to the constructor (for consistency and clarity with the base), you can apply the `public` keyword on the last prop to remove the assignment in the constructor.

```suggestion
  constructor(
    meta: TaskMetaData,
    reportComponentType: ReportComponentType = ReportComponentType.Table,
    public taskResult: DependencyResult[]
  ) {
    super(meta, reportComponentType);
  }
```

Reply:
You can likely apply this pattern to all the classes you've defined.